### ISSUE TITLE
Backport: upgrade from 2.2.3 to 2.2.4 in any MC apps `Target Projects *` is null

### ISSUE THREAD

**Issue by westlywright** [rancher/rancher#20857]
Backport rancher/rancher#20698

**Comment by alena1108** [comment]
Will be available in v2.2.5-rc2

**Comment by davidnuzik** [comment]
Version: v2.2.5-rc2

It is difficult to get into a state where we have projects without clusters.
In fact we were not able to determine a good way to do this.

I have tested and validated this should be working correctly. I didn't encounter any unexpected issues in the UI with my testing in this area. MC apps were smoke tested and we ensured in general the Target Projects drop down was working as expected and that mc apps were brought up correctly. HTTP/API/UI/apps look good.

**Comment by davidnuzik** [comment]
Reopened. We have a way to cause the issue where we have projects without clusters. Will cause this scenario and test again to fully validate.

**Comment by davidnuzik** [comment]
**Replication, Version: v2.2.4**
I created orphaned projects by creating eks clusters and deleting before ready.
I replicated the Target projects issue:

![image](https://user-images.githubusercontent.com/45179589/59529179-2693c300-8e95-11e9-9f28-c288980ea840.png)

**Validation, Version: v2.2.5-rc2**
I upgraded to v2.2.5-rc2 and we now filter out orphaned projects so the Target drop down now works correctly. I confirmed post-upgrade it works whether or not we have clusters up yet. When I bring additional clusters up the list updates as expected and I smoke tested multi-cluster apps and they all deployed correctly. There are no longer any web console errors as well. HTTP/API/Apps look correct and are created in all designated areas (projects I specify).

![image](https://user-images.githubusercontent.com/45179589/59529370-b0439080-8e95-11e9-8bd6-25c22037d5fc.png)

![image](https://user-images.githubusercontent.com/45179589/59529413-d1a47c80-8e95-11e9-8de0-c46fc7bae9ec.png)
